### ISSUE TITLE
Clarify the best practice to continue training after loading from a lora checkpoint

### ISSUE THREAD

**Issue by hanyin88** [huggingface/peft#1127]
### System Info

N/A

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder
- [ ] My own task or dataset (give details below)

### Reproduction

N/A

### Expected behavior

Hello,

I would like to clarify the best practice to continue training after loading from a lora checkpoint. After reading numerous posts I did not find an official recommendation, so figured to ask here. 

Let's save we have a saved lora checkpoint, and we would like to continue training.  So we have:
```
model = LlamaForSequenceClassification.from_pretrained(base_model)
config = LoraConfig(r=lora_r, lora_alpha=lora_alpha, target_modules=lora_target_modules )

```
From my research, there are two options here. The first appraoch is from[ alpaca-lora](https://github.com/tloen/alpaca-lora/blob/main/finetune.py):
```
model = get_peft_model(model, config)
adapters_weights = torch.load(checkpoint_name)
set_peft_model_state_dict(model, adapters_weights)
trainer = Trainer(
        model=model, ***)
```

The other approach is suggested in this [post](https://github.com/huggingface/peft/issues/859):
```
model = PeftModel.from_pretrained(model, adapters_weights, is_trainable=True)
trainer = Trainer(
        model=model, ***)
```

I am wondering is there any difference between the two approach? Seems to be either way would work. 

In addition with the second approach, I am not sure if these lines are necessary (suggested [here](https://github.com/huggingface/peft/issues/184) by @shaunheilee ):
```
from peft.tuners.lora import mark_only_lora_as_trainable
lora_model = PeftModel.from_pretrained(in_model, path, is_trainable=True)
mark_only_lora_as_trainable(lora_model)
```

Many thanks.

**Comment by younesbelkada** [comment]
Hi @hanyin88 
I advise you to use either the `PeftModel` interface through
```python
model = PeftModel.from_pretrained(model, adapters_weights, is_trainable=True)
```
or use the transformers + peft integration (available only from transformers>=4.34.0) where you would just need to do 
```python
model = AutoModelForCausalLM.from_pretrained(adapter_model_id, **kwargs)
```
Check out more about it [here](https://huggingface.co/docs/transformers/peft)

The first solution is equivalent to tbe solution you mentioned in alpaca-lora, but will do everything for you under the hood

**Comment by TapendraBaduwal** [comment]
@younesbelkada can you provide me the code to continue training after loading from a lora checkpoint?

**Comment by kodinkod** [comment]
it's work for UNet2DConditionModel ?

> Hi @hanyin88 I advise you to use either the `PeftModel` interface through
> 
> ```python
> model = PeftModel.from_pretrained(model, adapters_weights, is_trainable=True)
> ```
> 
> or use the transformers + peft integration (available only from transformers>=4.34.0) where you would just need to do
> 
> ```python
> model = AutoModelForCausalLM.from_pretrained(adapter_model_id, **kwargs)
> ```
> 
> Check out more about it [here](https://huggingface.co/docs/transformers/peft)
> 
> The first solution is equivalent to tbe solution you mentioned in alpaca-lora, but will do everything for you under the hood

**Comment by younesbelkada** [comment]
For using diffusers + peft please have a look at https://huggingface.co/docs/diffusers/main/en/tutorials/using_peft_for_inference#load-loras-for-inference or the diffusers example scripts. cc @sayakpaul

**Comment by sayakpaul** [comment]
This should be turned into Discussion I think.

**Comment by sayakpaul** [comment]
Let's move this to https://github.com/huggingface/peft/discussions

**Comment by BigDataMLexplorer** [comment]
> Hi @hanyin88 I advise you to use either the `PeftModel` interface through
> 
> ```python
> model = PeftModel.from_pretrained(model, adapters_weights, is_trainable=True)
> ```
> 
> or use the transformers + peft integration (available only from transformers>=4.34.0) where you would just need to do
> 
> ```python
> model = AutoModelForCausalLM.from_pretrained(adapter_model_id, **kwargs)
> ```
> 
> Check out more about it [here](https://huggingface.co/docs/transformers/peft)
> 
> The first solution is equivalent to tbe solution you mentioned in alpaca-lora, but will do everything for you under the hood

Hi, here -> (model = AutoModelForCausalLM.from_pretrained(adapter_model_id, **kwargs)) is adapter_model_id path to 
 checkpoint? And in in Trainer.train(resume_from_checkpoint) I sholud use resume_from_checkpoint = adapter_model_id?
